stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -82,7 +82,28 @@ function(_add_target_variant_c_compile_link_flags)
       DEPLOYMENT_VERSION "${DEPLOYMENT_VERSION}")
     list(APPEND result "-target" "${target}")
     if(target_variant)
-      list(APPEND result "-target-variant" "${target_variant}")
+      # Check if the C compiler supports `-target-variant` flag
+      # TODO (etcwilde): This is a massive hack to deal with the fact that we
+      # are lying to cmake about what compiler is being used. Normally we could
+      # use `check_cxx_compiler_flag` here. Unfortunately, that uses a different
+      # compiler since we swap out the C compiler part way through the build.
```

Comment:
I really wish that we would stop doing this ...

Reply:
I do too. I do too.